usecase/usecase-realsense-xwalk-tests/samples/PhotoSet/js/main.js

```diff
@@ -0,0 +1,209 @@
+ARE DISCLAIMED. IN NO EVENT SHALL INTEL CORPORATION BE LIABLE FOR ANY DIRECT,
+INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,
+EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*/
+
+var photoCapture;
+var imgContext;
+var currentPhoto;
+var width;
+var height;
+
+window.onload = function() {
+  width = $("#imgPreview")[0].width;
+  height = $("#imgPreview")[0].height;
+  imgContext = $("#imgPreview")[0].getContext("2d");
+  $("#takePhoto").click(takePhoto);
+  $("#setContainerImage").click(setContainerImage);
+  $("#setColorImage").click(setColorImage);
+  $("#setDepthImage").click(setDepthImage);
+  $("#setRawDepthImage").click(setRawDepthImage);
+  getStream();	
+}
+
+function getStream() {
+  var constraints = {video: {
+    width: 320,
+    height: 240}
+  };	
+  navigator.getUserMedia(constraints, function successCallback(stream) {
+    $("#preview")[0].src = URL.createObjectURL(stream);
+    photoCapture = new realsense.DepthEnabledPhotography.PhotoCapture(stream);
+    $("#log").text("init success");
+  }, function(ex) {
+    $("#log").text("access video stream fail.");
+  });	
+}
+
+function takePhoto() {
+  photoCapture.takePhoto()
+    .then(function(photo) {
+      currentPhoto = photo;
+      return currentPhoto.queryContainerImage();
+    })
+    .then(function(image) {
+      fillCanvas(image, false);
+    })
+    .catch(errorCallback);	
+}
+
+function setContainerImage(){
+  if(currentPhoto == null) {
+    $("#log").text("please capture photo first!");
+    return;
+  }
+  photoCapture.takePhoto()
+    .then(function(photo) {
+      return photo.queryContainerImage();
+    })
+    .then(function(image) {
+      return currentPhoto.setContainerImage(image);
+    })
+    .then(function() {
+      $("#log").text("set container image succeed!");
+      return currentPhoto.queryContainerImage();
+    })
+    .then(function(image) {
+      fillCanvas(image, false);
+    })
+    .catch(errorCallback);
+}
+
+function setColorImage() {
+  if(currentPhoto == null) {
+    $("#log").text("please capture photo first!");
+    return;
+  }
+  photoCapture.takePhoto()
+    .then(function(photo) {
+      return photo.queryImage();
+    })
+    .then(function(image) {
+      return currentPhoto.setColorImage(image);
+    })
+    .then(function() {
+      $("#log").text("set color image succeed!");
+      return currentPhoto.queryImage();
+    })
+    .then(function(image) {
+      fillCanvas(image, false);
+    })
+    .catch(errorCallback);	
+}
+
+function setDepthImage() {
+  if(currentPhoto == null) {
+    $("#log").text("please capture photo first!");
+    return;
+  }
+  photoCapture.takePhoto()
```

Comment:
Yes, if i only do queryDepth() operation, it can return successfully,
 if i call photoCapture.takePhoto() before doing queryDepth operation, then it will get AbortError exception.